### PR TITLE
Improve pagination API

### PR DESCRIPTION
**Pull request by Aaron1011**
SpongeAPI | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/434)

This PR improves the usability of the Pagination API.

Changes:
* Added `PaginationList`, an immutable version of `PaginationBuilder`. A `PaginationList` is obtained from a `PaginationBuilder`.
* Made `PaginationBuilder` a `ResettableBuilder`
* Changed `PaginationBuilder` to use`MessageReciever`instead of `CommandSource`, and added methods for using `MessageChannel`.

This resolves https://github.com/SpongePowered/SpongeAPI/issues/1010

### CODE REVIEW

File: src/main/java/org/spongepowered/api/service/pagination/PaginationBuilder.java [hunk 1]
```diff
@@ -25,14 +25,16 @@
...
+import org.spongepowered.api.text.channel.MessageReceiver;
+import org.spongepowered.api.util.ResettableBuilder;
 
 import java.util.List;
 
 /**
  * Builds a paginated output for an iterable of {@link Text}s.
  */
-public interface PaginationBuilder {
+public interface PaginationBuilder extends ResettableBuilder<PaginationList, PaginationBuilder> {
```
Comment (JBYoshi) [inline src/main/java/org/spongepowered/api/service/pagination/PaginationBuilder.java]: How come this isn't an inner interface of `PaginationList`?

File: src/main/java/org/spongepowered/api/service/pagination/PaginationBuilder.java [hunk 2]
```diff
@@ -87,10 +89,27 @@
...
      */
-    void sendTo(CommandSource source);
+    void sendTo(MessageReceiver source);
+
+    /**
+     * Send the constructed pagination list to the specified {@link MessageChannel}.
+     * @param channel
+     */
+    default void sendTo(MessageChannel channel) {
+        for (MessageReceiver receiver: channel.getMembers()) {
```
Comment (phroa) [inline src/main/java/org/spongepowered/api/service/pagination/PaginationBuilder.java]: Space before the `:`

File: src/main/java/org/spongepowered/api/service/pagination/PaginationList.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+     * @param source The source to send to
+     */
+    void sendTo(MessageReceiver source);
+
+    /**
+     * Send the constructed pagination list to the specified {@link MessageChannel}.
+     * @param channel
+     */
+    default void sendTo(MessageChannel channel) {
+        for (MessageReceiver receiver: channel.getMembers()) {
```
Comment (phroa) [inline src/main/java/org/spongepowered/api/service/pagination/PaginationList.java]: Space before the colon

**Comment by Feeliiix** [comment]
@Aaron1101 it would be very nice if you add a way to style the paddingString.

File: src/main/java/org/spongepowered/api/service/pagination/PaginationCalculator.java [hunk 4]
```diff
@@ -26,11 +26,12 @@
 
 import org.spongepowered.api.text.Text;
 import org.spongepowered.api.command.CommandSource;
```
Comment (phroa) [inline src/main/java/org/spongepowered/api/service/pagination/PaginationCalculator.java]: I think you can delete this import now.

**Pull request closed without merging**